src/SME.SGP.Dominio.Servicos/ServicoNotificacaoFrequencia.cs

```diff
@@ -740,14 +743,54 @@ private DateTime DiaRetroativo(DateTime data, int nrDias)
 
             return dataRetorno;
         }
-
+        private bool GerarNotificacaoExtemporanea(bool periodoAberto, bool parametroAtivo)
+        {
+            if (periodoAberto)
+                return false;
+            else if (parametroAtivo && !periodoAberto)
+                return true;
+            else if (!parametroAtivo && !periodoAberto)
+                throw new NegocioException("Compensação de ausência não permitida, É necessário que o período esteja aberto");
+            else if (!parametroAtivo)
+                return false;
```

Comment:
Se não entrar na primeira condição já quer dizer que o periodo não esta aberto, não precisa checar novamente por causa do else
```suggestion
            else if (parametroAtivo)
                return true;
            else
                throw new NegocioException("Compensação de ausência não permitida, É necessário que o período esteja aberto");
```